pkg/descheduler/evictions/evictions.go

```diff
@@ -187,6 +152,66 @@ func evictPod(ctx context.Context, client clientset.Interface, pod *v1.Pod, poli
 	return err
 }
 
+type constraint func(pod *v1.Pod) error
+
+type evictable struct {
```

Comment:
It seems odd that this is a private type but all the functions that work with it are public. Might be better as a public type, and then if we add more options to it in the future, it could be declared as a literal

I'm not sure on the name, I think something like `EvictOptions` might be clearer. wdyt?

Reply:
That was the idea. Keep the actual implementation hidden and just rely on a single `IsEvictable` invocation. I have created `Options` struct instead of having `WithPriorityThreshold` as a method.